packages/language-server/src/ui5-model.ts

```diff
@@ -11,33 +18,104 @@ import {
 } from "@ui5-language-assistant/semantic-model";
 import { Fetcher } from "../api";
 import { getLogger } from "./logger";
+import globby from "globby";
+import { parse } from "yaml";
+
+const DEFAULT_UI5_FRAMEWORK = "sapui5";
+const DEFAULT_UI5_VERSION = "1.71.49";
 
-const DEFAULT_UI5_VERSION = "1.71.14";
+const UI5_FRAMEWORK_CDN_BASE_URL = {
+  openui5: "https://sdk.openui5.org/",
+  sapui5: "https://ui5.sap.com/",
+};
 
 export async function getSemanticModel(
-  modelCachePath: string | undefined
+  modelCachePath: string | undefined,
+  workspacePath: string | undefined
 ): Promise<UI5SemanticModel> {
-  return getSemanticModelWithFetcher(fetch, modelCachePath);
+  return getSemanticModelWithFetcher(fetch, modelCachePath, workspacePath);
 }
 
 // This function is exported for testing purposes (using a mock fetcher)
 export async function getSemanticModelWithFetcher(
```

Comment:
can some logic be extracted? this function is now 140 lines long...

Reply:
Yes, but I would do so with the next iteration of the language assistant making it work for mono-repos